src/index.js

```diff
@@ -1,390 +1,465 @@
+  for (var i in b) out[i] = b[i]
 
-  function render() {
-    skipRender = !skipRender
+  return out
+}
 
-    var node = resolveNode(view)
+var flatten = function(arr) {
+  return arr.reduce(function(out, obj) {
+    return out.concat(
+      !obj || obj === true
+        ? false
+        : typeof obj[0] === "function"
+        ? [obj]
+        : flatten(obj)
+    )
+  }, EMPTY_ARRAY)
+}
 
-    if (container && !skipRender) {
-      rootElement = patch(container, rootElement, oldNode, (oldNode = node))
-    }
+var isSameAction = function(a, b) {
+  return isArray(a) && isArray(b) && a[0] === b[0] && typeof a[0] === "function"
+}
 
-    isRecycling = false
+var shouldRestart = function(a, b) {
+  for (var k in merge(a, b)) {
+    if (a[k] === b[k] || isSameAction(a[k], b[k])) b[k] = a[k]
+    else return true
+  }
+}
 
-    while (lifecycle.length) lifecycle.pop()()
+var patchSub = function(sub, newSub, dispatch) {
+  for (var i = 0, a, b, out = []; i < sub.length || i < newSub.length; i++) {
+    a = sub[i]
+    out.push(
+      (b = newSub[i])
+        ? !a || b[0] !== a[0] || shouldRestart(b[1], a[1])
+          ? [b[0], b[1], b[0](b[1], dispatch), a && a[2]()]
+          : a
+        : a && a[2]()
+    )
   }
+  return out
+}
 
-  function scheduleRender() {
-    if (!skipRender) {
-      skipRender = true
-      setTimeout(render)
+var updateProperty = function(element, name, value, newValue, eventCb, isSvg) {
+  if (name === "key") {
+  } else if (name === "style") {
+    for (var i in merge(value, newValue)) {
+      var style = newValue == null || newValue[i] == null ? "" : newValue[i]
+      if (i[0] === "-") {
+        element[name].setProperty(i, style)
+      } else {
+        element[name][i] = style
+      }
     }
+  } else if (name[0] === "o" && name[1] === "n") {
+    if (
+      !((element.events || (element.events = {}))[
+        (name = name.slice(2))
+      ] = newValue)
+    ) {
+      element.removeEventListener(name, eventCb)
+    } else if (!value) {
+      element.addEventListener(name, eventCb)
+    }
+  } else if (name !== "list" && !isSvg && name in element) {
+    element[name] = newValue == null ? "" : newValue
+  } else if (
+    newValue == null ||
+    newValue === false ||
+    (name === "class" && !(newValue = createClass(newValue)))
+  ) {
+    element.removeAttribute(name)
+  } else {
+    element.setAttribute(name, newValue)
   }
+}
 
-  function clone(target, source) {
-    var out = {}
-
-    for (var i in target) out[i] = target[i]
-    for (var i in source) out[i] = source[i]
+var removeElement = function(parent, node) {
+  parent.removeChild(node.element)
+}
 
-    return out
+var createElement = function(node, eventCb, isSvg) {
+  var element =
+    node.type === TEXT_NODE
+      ? document.createTextNode(node.name)
+      : (isSvg = isSvg || node.name === "svg")
+      ? document.createElementNS("http://www.w3.org/2000/svg", node.name)
+      : document.createElement(node.name)
```

Comment:
Can we support customized built-in elements here? For example:
```
<p is="word-count">
```
and
```
document.createElement("p", { is: "word-count" })
```

Reply:
@dennis84 Hyperapp sets attributes/properties after creating the element. This code:

```jsx
<p is="word-count">
```

...creates a `p` element with an `is` attribute exactly as declared. Is that what you needed?

<img width="636" alt="Screen Shot 2019-04-15 at 21 44 55" src="https://user-images.githubusercontent.com/56996/56133593-bfa67a80-5fc7-11e9-9b82-26d54d5b2bac.png">

Reply:
Ah, that doesn't work for custom elements. As Far as I know, you have to pass the "is" attribute as options to the createElement function.

If you run this in the conosle nothing is logged:
```javascript
class WordCount extends HTMLParagraphElement {
  connectedCallback() {
    console.log('AAAA')
  }
}

window.customElements.define('word-count', WordCount, {extends: 'p'})

var p = document.createElement('p')//, {is: 'word-count'})
p.setAttribute('is', 'word-count')
p.appendChild(document.createTextNode('test'))
document.body.appendChild(p)
```

Reply:
@dennis84 Does React/Inferno/Preact/etc. handle this?

Reply:
I could only find it in React and Mithril:
- https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOMComponent.js#L429
- https://github.com/MithrilJS/mithril.js/blob/master/mithril.js#L554

Reply:
@dennis84 Awesome, thanks. I'll figure out what's the best way to add this after V2 is out. Thank you.